src/_pytest/mark/structures.py

```diff
@@ -111,7 +111,19 @@ def _for_parametrize(cls, argnames, argvalues, func, config):
         ]
         del argvalues
 
-        if not parameters:
+        if parameters:
+            # check all parameter sets have the correct number of values
+            for param in parameters:
+                if len(param.values) != len(argnames):
+                    raise ValueError(
+                        'In "parametrize" the number of values ({}) must be '
+                        "equal to the number of names ({})".format(
```

Comment:
At first glance I would agree with you, but when you look at the actual message I think it is actually fine:

```
ValueError: In "parametrize" the number of values ((1, 2, 3)) must be equal to the number of names (['x', 'y'])
```

Showing all values seems like the right choice too, because we can't assume which of the arguments are missing or are in excess, because the user might have removed one of the parameters from the middle of the argument list.

Also it is hard to tell from the diff, but I moved this check verbatim from here:

https://github.com/pytest-dev/pytest/blob/6c3713226c1c619196ddc0fe870ea4fa4f5253c3/src/_pytest/python.py#L924-L935